cupyx/scipy/sparse/construct.py

```diff
@@ -88,6 +88,234 @@ def spdiags(data, diags, m, n, format=None):
     return dia.dia_matrix((data, diags), shape=(m, n)).asformat(format)
 
 
+def _compressed_sparse_stack(blocks, axis):
+    """
+    Stacking fast path for CSR/CSC matrices
+    (i) vstack for CSR, (ii) hstack for CSC.
+    """
+    other_axis = 1 if axis == 0 else 0
+    data = cupy.concatenate([b.data for b in blocks])
+    constant_dim = blocks[0].shape[other_axis]
+    idx_dtype = get_index_dtype(arrays=[b.indptr for b in blocks],
+                                maxval=max(data.size, constant_dim))
+    indices = cupy.empty(data.size, dtype=idx_dtype)
+    indptr = cupy.empty(sum(b.shape[axis]
+                            for b in blocks) + 1, dtype=idx_dtype)
+    last_indptr = idx_dtype(0)
+    sum_dim = 0
+    sum_indices = 0
+    for b in blocks:
+        if b.shape[other_axis] != constant_dim:
+            raise ValueError(
+                'incompatible dimensions for axis %d' % other_axis)
+        indices[sum_indices:sum_indices+b.indices.size] = b.indices
+        sum_indices += b.indices.size
+        idxs = slice(sum_dim, sum_dim + b.shape[axis])
+        indptr[idxs] = b.indptr[:-1]
+        indptr[idxs] += last_indptr
+        sum_dim += b.shape[axis]
+        last_indptr += b.indptr[-1]
+    indptr[-1] = last_indptr
+    if axis == 0:
+        return csr.csr_matrix((data, indices, indptr),
+                              shape=(sum_dim, constant_dim))
+    else:
+        return csc.csc_matrix((data, indices, indptr),
+                              shape=(constant_dim, sum_dim))
+
+
+def hstack(blocks, format=None, dtype=None):
+    """
+    Stack sparse matrices horizontally (column wise)
+
+    Args:
+        blocks (sequence of cupyx.scipy.sparse.spmatrix):
+            sparse matrices to stack
+
+        format (str):
+            sparse format of the result (e.g. "csr")
+            by default an appropriate sparse matrix format is returned.
+            This choice is subject to change.
+        dtype (dtype, optional):
+            The data-type of the output matrix.  If not given, the dtype is
+            determined from that of `blocks`.
```

Comment:
We need ``` `` ``` to correctly render the code fragments.
Please fix other docstring as well

```suggestion
            determined from that of ``blocks``.
```